directory/directory_test.go

```diff
@@ -0,0 +1,71 @@
+package directory
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCanonicalDockerReference(t *testing.T) {
+	dest := NewDirImageDestination("/path/to/somewhere")
+	_, err := dest.CanonicalDockerReference()
+	assert.NotNil(t, err)
+}
+
+func TestGetPutManifest(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-manifest")
+	assert.Nil(t, err)
+	dest := NewDirImageDestination(tmp)
+	err = dest.PutManifest([]byte("test-manifest"))
+	assert.Nil(t, err)
+	src := NewDirImageSource(tmp)
+	m, mt, err := src.GetManifest(nil)
+	assert.Nil(t, err)
+	assert.Equal(t, []byte("test-manifest"), m)
```

Comment:
The `[]byte(…)` would be better as a constant if possible, or at least a variable (private to the function), to make the identity obvious.